nemo/utils/app_state.py

```diff
@@ -12,11 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass
+from threading import Lock
+from typing import Optional
+
 from nemo.utils.metaclasses import Singleton
 
 
+@dataclass()
+class ModelMetadataRegistry:
+    guid: str
+    gidx: int
```

Comment:
What's `gidx` used for?

Reply:
Unused as of now, just a global index of the model in the scope of AppState.